### ISSUE TITLE
Program using cgo are killed at launch

### ISSUE THREAD

**Issue by maxence-charriere** [golang/go#19841]
Please answer these questions before submitting your issue. Thanks!

### What version of Go are you using (`go version`)?
```
go version go1.8 darwin/amd64
```

### What operating system and processor architecture are you using (`go env`)?
```
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/maxence/Dev/go"
GORACE=""
GOROOT="/usr/local/Cellar/go/1.8/libexec"
GOTOOLDIR="/usr/local/Cellar/go/1.8/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/07/tj_2fpcd5lv00gj1shc_kgqw0000gn/T/go-build751688862=/tmp/go-build -gno-record-gcc-switches -fno-common"
CXX="clang++"
CGO_ENABLED="1"
PKG_CONFIG="pkg-config"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
```

### What did you do?
```go
package main

import "C"
import "fmt"

func main() {
	fmt.Println("Hello maxoo")
}
```

### What did you expect to see?
```
github.com/maxence-charriere/test
▶ ./test
Hello maxoo
```

### What did you see instead?
```
github.com/maxence-charriere/test
▶ ./test
[1]    16391 killed     ./test
```

**Comment by ianlancetaylor** [comment]
Dup of #19734?

**Comment by maxence-charriere** [comment]
Look like the other topic is pretty random depending on the c library used.
Here I just use ```go build``` and there is just an ```import "C"```.
There is no use of any c functions.

I also tried using a simple printf, the result was the same.

This started to occur since I updated to go 1.8.

using ```go build -ldflags -s``` solve the problem but i doubt this should be the default behavior since it's not documented anywhere.
